include/deal.II/matrix_free/fe_remote_evaluation.h

```diff
@@ -0,0 +1,768 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+#ifndef dealii_matrix_free_fe_remote_evaluation_h
+#define dealii_matrix_free_fe_remote_evaluation_h
+
+#include <deal.II/base/mpi_remote_point_evaluation.h>
+
+#include <deal.II/matrix_free/fe_evaluation.h>
+#include <deal.II/matrix_free/fe_point_evaluation.h>
+#include <deal.II/matrix_free/matrix_free.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+#include <variant>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace internal
+{
+  /**
+   * A class that stores values and/or gradients at quadrature points
+   * corresponding to a FEEvaluationType.
+   */
+  template <int dim, int n_components, typename value_type_>
+  struct PrecomputedFEEvaluationData
```

Comment:
What about calling the class `PrecomputedEvaluationData`? It is independet of `FEEvaluation`.